src/src/com/microsoft/aad/adal/BlobContainer.java

```diff
@@ -0,0 +1,129 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import java.io.Serializable;
+
+public class BlobContainer implements Serializable {
+
+    /**
+     * universal version identifier for BlobContainer class
+     */
+    private static final long serialVersionUID = 5722517776407773544L;
+
+    /**
+     * the BlobContainer stores the FRT tokenCacheItem 
+     * of the given user
+     */
+    private TokenCacheItem tokenItem;
+    
+    private String familyRefreshToken;
+    
+    /**
+     * Construct default blob container.
+     */
+    public BlobContainer() {
+        tokenItem = new TokenCacheItem();
+    }
+    
+    /**
+     * Check if the cache item contains family refresh token
+     * if true, value the tokenItem with this cache item
+     * @param item
+     * @throws AuthenticationException 
+     */
+    public BlobContainer(TokenCacheItem tokenItem) throws AuthenticationException {
+        //only FRT is stored here 
+        if(tokenItem == null) {
+            throw new IllegalArgumentException("invalid input TokenCacheItem");
+        } else if (tokenItem != null 
+                && StringExtensions.IsNullOrBlank(tokenItem.getFamilyClientId())) {
+            throw new AuthenticationException(
+                    ADALError.FAMILY_REFRESH_TOKEN_NOT_FOUND, 
+                    "No family refresh token is found from the cache item"
+                    );
+        } else if (tokenItem != null 
+                && !StringExtensions.IsNullOrBlank(tokenItem.getFamilyClientId())
+                && tokenItem.getFamilyClientId().equalsIgnoreCase("1")) {
+            this.tokenItem = tokenItem;
+            this.familyRefreshToken = tokenItem.getRefreshToken();
+        }
+    }
+    
+    public TokenCacheItem getTokenItem() {
+        return tokenItem;
+    }
+
+    public void setTokenItem(TokenCacheItem tokenItem) {
+        this.tokenItem = tokenItem;
+    }    
+    
+    /**
+     * Get the family refresh token from the tokenItem
+     * @return family refresh token
+     * @throws AuthenticationException 
+     */
+    public String getFamilyRefreshToken() throws AuthenticationException {
+        return this.familyRefreshToken;
+    }
+    
+    public void setFamilyRefreshToken(String familyRefreshToken) {
+        this.familyRefreshToken = familyRefreshToken;
+    }
+    
+    public void setFamilyRefreshToken(TokenCacheItem tokenItem) throws AuthenticationException {
```

Comment:
Wait - now I'm totally confused. Why do we have `setToken` & `setFamilyRefreshToken`